src/twisted/logger/test/test_json.py

```diff
@@ -280,6 +280,40 @@ def test_observeWritesEmptyRecordSeparator(self):
         self.assertObserverWritesJSON(recordSeparator=u"")
 
 
+    def test_failureFormatting(self):
+        """
+        A L{FileLogObserver} created by L{jsonFileLogObserver} writes failures
+        serialized as JSON text to a file when it observes events.
+        """
+        io = StringIO()
+        publisher = LogPublisher()
+        before = []
+        publisher.addObserver(before.append)
+        publisher.addObserver(jsonFileLogObserver(io))
+        logger = Logger(observer=publisher)
+        try:
+            1 / 0
+        except:
+            logger.failure("failed as expected")
+        reader = StringIO(io.getvalue())
+        events = list(eventsFromJSONLogFile(reader))
+        self.assertEqual(len(events), 1)
+        self.assertEqual(len(before), 1)
+        [event] = events
+        [beforeEvent] = before
+        self.assertIn("log_failure", event)
+        f = event["log_failure"]
+        b = beforeEvent["log_failure"]
+        self.assertIsInstance(f, Failure)
+        deserializedTraceback = f.getTracebackObject()
+        previousTraceback = b.getTracebackObject()
+        def checkOneTraceback(tb):
+            self.assertEqual(tb.tb_frame.f_code.co_filename.rstrip("co"),
```

Comment:
An additional check that confirms the `f_code.co_name` is the same as the test method's would be nice.  I guess you didn't include that, though, because it would make the code depend on the method's name?